android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -336,6 +340,11 @@ class GeckoEngineSession(
             return false
         }
 
+        // TODO fetch the correct value for enableDesktopMode
```

Comment:
Argh, this comment is the place where you've discovered that this solution isn't enough. It looks like we should have GeckoView save and restore the state in their `EngineSessionState` (`GeckoSessionState`) which we can then apply here.

Please file a bug in the GeckoView component to add support for restoring desktop mode. I think your change today can still work with some tweaks.

A possible solution is to set the desktop mode _before_ the `restoreState` is called. This happens in the `CreateEngineSessionMiddleware` - specifically above [this line](https://searchfox.org/mozilla-mobile/rev/9f5ba3d5f8fb052d07447bb3b406d4a3b8ca2d82/firefox-android/android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/CreateEngineSessionMiddleware.kt#119) where we can set the value before instructing the `EngineSession` to load.

That might suffice for now.